### PR TITLE
Moving reload_analyzers endpoint to xpack

### PR DESCRIPTION
**Pull request by cbuescher**
This change moves the new "_reload_search_analyzers" endpoint into x-pack core so reloading of analyzers is available under the Basic license.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/TransportReloadAnalyzersAction.java [hunk 1]
```diff
@@ -1,27 +1,12 @@
...
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
  */
-
-package org.elasticsearch.action.admin.indices.reloadanalyzer;
+package org.elasticsearch.xpack.core.rest.action;
```
Comment (jimczi) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/TransportReloadAnalyzersAction.java]: Can you move this file to `org.elasticsearch.xpack.core.action` ? The `rest.action` package is for the rest handlers.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/ReloadAnalyzerRequestBuilder.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * Builder for reloading of analyzers
+ */
+public class ReloadAnalyzerRequestBuilder
+        extends BroadcastOperationRequestBuilder<ReloadAnalyzersRequest, ReloadAnalyzersResponse, ReloadAnalyzerRequestBuilder> {
+
+    public ReloadAnalyzerRequestBuilder(ElasticsearchClient client, ReloadAnalyzerAction action, String... indices) {
+        super(client, action, new ReloadAnalyzersRequest(indices));
+    }
+
+}
```
Comment (jimczi) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/ReloadAnalyzerRequestBuilder.java]: This class doesn't seem to be used, it is needed ?

**Comment by cbuescher** [comment]
@elasticmachine update branch

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/ReloadAnalyzersResponse.java [hunk 3]
```diff
@@ -1,22 +1,9 @@
...
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
  */
-package org.elasticsearch.action.admin.indices.reloadanalyzer;
+package org.elasticsearch.xpack.core.rest.action;
```
Comment (jimczi) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/ReloadAnalyzersResponse.java]: same here

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/ReloadAnalyzersRequest.java [hunk 4]
```diff
@@ -1,23 +1,9 @@
...
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
  */
-
-package org.elasticsearch.action.admin.indices.reloadanalyzer;
+package org.elasticsearch.xpack.core.rest.action;
```
Comment (jimczi) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/ReloadAnalyzersRequest.java]: I think it should move to org.elasticsearch.xpack.core.action or as a static inner class of TransportReloadAnalyzersAction if we don't want to have a dedicated file for this.

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/rest/action/ReloadSynonymAnalyzerTests.java [hunk 5]
```diff
@@ -1,31 +1,20 @@
...
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
  */
 
-package org.elasticsearch.analysis.common;
+package org.elasticsearch.xpack.core.rest.action;
```
Comment (jimczi) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/rest/action/ReloadSynonymAnalyzerTests.java]: This is not a rest test so I think it should also move to `org.elasticsearch.xpack.core.action`

**Review by jimczi — COMMENTED** [review]
Thanks @cbuescher , I left some comments

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/ReloadAnalyzerRequestBuilder.java (hunk 2 above)
Comment (cbuescher) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rest/action/ReloadAnalyzerRequestBuilder.java]: Right, it was used before with some other convenience classes for the Node client that I removed. Will also remove this one.

**Comment by cbuescher** [comment]
@elasticmachine update branch

**Comment by cbuescher** [comment]
@jimczi moved the classes you mentioned, also looks like the CI gods are smiling upon us now.

**Review by jimczi — APPROVED** [review]
LGTM

**Pull request merged**